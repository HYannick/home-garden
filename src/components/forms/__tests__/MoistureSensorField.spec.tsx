import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import { Formik } from 'formik';
import { renderWithTheme } from '../../../theme-wrapper';
import MoistureSensorInput from '../MoistureSensorField';

afterEach(cleanup);
describe('MoistureSensorField', () => {
  const props = {
    setFieldValue: jest.fn(),
    resetField: jest.fn(),
    values: {
      name: '',
      picture: null,
      last_watering_date: '',
      has_moisture_sensor: false,
      sensor_id: '',
      watering_frequency: 2,
      need_watering_frequency: false,
      custom_name: '',
    },
    testId: 'test',
    errors: {},
    t: jest.fn(),
  };
  it('should render the MoistureSensorField properly', () => {
    const { container } = renderWithTheme(
      <Formik
        initialValues={props.values}
        render={() => <MoistureSensorInput {...props}/>}
        onSubmit={jest.fn()}/>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should show watering_frequency if need_watering_frequency is enabled', () => {
    const newProps = {
      ...props,
      values: {
        ...props.values,
        need_watering_frequency: true,
      }
    };
    const { getByTestId } = renderWithTheme(
      <Formik
        initialValues={props.values}
        render={() => <MoistureSensorInput {...newProps}/>}
        onSubmit={jest.fn()}/>,
    );

    expect(getByTestId(/^need_watering_frequency/)).toBeTruthy();
  });
});
