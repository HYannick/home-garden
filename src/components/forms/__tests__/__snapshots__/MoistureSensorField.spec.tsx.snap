// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoistureSensorField should render the MoistureSensorField properly 1`] = `
<div>
  <div
    class="css-wwcne1-MoistureSensorInput"
  />
  <div
    class="css-1diq458"
  >
    <div
      class="css-1r43qjs-MoistureSensorInput"
    >
      <label
        class="css-npa6bc"
        for="sensor_id"
      />
      <input
        class="css-1gexmoh"
        id="sensor_id"
        name="sensor_id"
        type="text"
      />
    </div>
  </div>
  <div
    class="css-1diq458"
  >
    <label
      class="css-npa6bc"
      for="need_watering_frequency"
    >
      Do you want to set a watering frequency?
    </label>
    <label
      class="css-qwug7p"
      for="need_watering_frequency"
    >
      <input
        type="checkbox"
      />
      <input
        class="css-x6ytko"
        id="need_watering_frequency"
        name="need_watering_frequency"
        type="checkbox"
      />
      <span
        class="slider round css-843eyc"
      />
    </label>
  </div>
</div>
`;
